controllers/gpuaddon/clusterpolicy_resource_reconciler.go

```diff
@@ -124,6 +124,27 @@ func (r *ClusterPolicyResourceReconciler) setDesiredClusterPolicy(
 		Enabled: &enabled,
 	}
 
+	if gpuAddon.Spec.DriverVersion != "" {
```

Comment:
Which of these should take precedence? The driver image or the driver version? Maybe an error should be thrown if both are given and they are conflicting.

Reply:
The `GPUAddon.Spec.DriverVersion` is the field, which should take precedence, as this is what the user of the GPU Add-on can manually edit to enforce the desired driver version at any point. 

If the `GPUAddon.Spec.DriverVersion` field is not set by the user, then we pin a driver version ourselves, by looking at the GPU operator CSV related images array and using the `driver-image` we find there, as the default pinned driver version.

The important part is that, at any point, we let the users choose the driver version explicitly (by setting the `GPUAddon.Spec.DriverVersion` field). Unless they don't want/need to (by removing the `GPUAddon.Spec.DriverVersion` field), in which case we try to find the most suitable driver version ourselves and pin that one.

@empovit does this answer your question? Did I understand correctly your concerns?